synfig-studio/src/gui/render.cpp

```diff
@@ -339,39 +371,104 @@ RenderSettings::on_render_pressed()
 		catch(std::runtime_error& x)
 		{
 			canvas_interface_->get_ui_interface()->error(_("Unable to determine proper target from filename."));
-			return;
+			return false;
 		}
 	}
 
 	if(filename.empty() && calculated_target_name!="null")
 	{
 		canvas_interface_->get_ui_interface()->error(_("A filename is required for this target"));
-		return;
+		return false;
 	}
-
-	hide();
-
-	render_passes.clear();
-	if (toggle_extract_alpha.get_active())
+	
+	String full_name(filename); //include possible sequence_separator + time for image sequences
+	String extension(filename_extension(filename));
+	bool ext_multi_file = false; //output target is an image sequence
+	int n_frames_overwrite = 0;
+	
+	//Retrieve current render settings
+	RendDesc rend_desc(widget_rend_desc.get_rend_desc());
```

Comment:
It could be a const reference instead of copy.

Reply:
not really sure how to do that